freqtrade/optimize/hyperopt.py

```diff
@@ -170,57 +170,43 @@ def log_trials_result(self) -> None:  # noqa: C901
-
-            if self.has_space('sell'):
-                result_dict['params'].update(self.space_params(params, 'sell'))
-
-            if self.has_space('roi'):
+        else:
+            self._params_pretty_print(params, 'buy', "Buy hyperspace params:")
+            self._params_pretty_print(params, 'sell', "Sell hyperspace params:")
+            self._params_pretty_print(params, 'roi', "ROI table:")
+            self._params_pretty_print(params, 'stoploss', "Stoploss:")
+            self._params_pretty_print(params, 'trailing', "Trailing stop:")
+
+    def _params_update_for_json(self, result_dict, params, space: str):
+        if self.has_space(space):
+            space_params = self.space_params(params, space)
+            if space in ['buy', 'sell']:
+                result_dict.setdefault('params', {}).update(space_params)
+            elif space == 'roi':
                 # Convert keys in min_roi dict to strings because
                 # rapidjson cannot dump dicts with integer keys...
                 # OrderedDict is used to keep the numeric order of the items
                 # in the dict.
                 result_dict['minimal_roi'] = OrderedDict(
-                    (str(k), v) for k, v in self.custom_hyperopt.generate_roi_table(params).items()
+                    (str(k), v) for k, v in space_params.items()
                 )
+            else:  # 'stoploss', 'trailing'
+                result_dict.update(space_params)
 
-            if self.has_space('stoploss'):
-                result_dict.update(self.space_params(params, 'stoploss'))
-
-            if self.has_space('trailing'):
-                result_dict.update(self.space_params(params, 'trailing'))
-
-            print(rapidjson.dumps(result_dict, default=str, number_mode=rapidjson.NM_NATIVE))
-        else:
-            if self.has_space('buy'):
-                print('Buy hyperspace params:')
-                pprint(self.space_params(params, 'buy', 5), indent=4)
-
-            if self.has_space('sell'):
-                print('Sell hyperspace params:')
-                pprint(self.space_params(params, 'sell', 5), indent=4)
-
-            if self.has_space('roi'):
-                print("ROI table:")
-                # Round printed values to 5 digits after the decimal point
-                pprint(round_dict(self.custom_hyperopt.generate_roi_table(params), 5), indent=4)
-
-            if self.has_space('stoploss'):
-                print(f"Stoploss:")
-                pprint(self.space_params(params, 'stoploss', 5), indent=4)
-
-            if self.has_space('trailing'):
-                print('Trailing stop:')
-                pprint(self.space_params(params, 'trailing', 5), indent=4)
+    def _params_pretty_print(self, params, space: str, header: str):
+        if self.has_space(space):
+            space_params = self.space_params(params, space, 5)
+            print(header)
+            pprint(space_params, indent=4)
```

Comment:
Now i have to confess, this looks pretty neat :+1:

Reply:
it was cleaner for a reader before, imho